src/cpp/transport/tcp/RTCPMessageManager.cpp

```diff
@@ -428,43 +441,50 @@ ResponseCode RTCPMessageManager::processBindConnectionRequest(
     //logError(DEBUG, "Receive Connection Request with locator: " << IPLocator::to_string(request.transportLocator())
     //    << " and will respond with our locator: " << response.locator());
 
-    ResponseCode code = p_channel_resource->process_bind_request(request.transportLocator());
-    sendData(p_channel_resource, BIND_CONNECTION_RESPONSE, transaction_id, &payload, code);
+    ResponseCode code = channel->process_bind_request(request.transportLocator());
+
+    if(RETCODE_OK == code)
+    {
+        // In server side, at this moment, the channel has to be included in the map.
+        mTransport->bind_socket(channel);
+    }
+
+    sendData(channel, BIND_CONNECTION_RESPONSE, transaction_id, &payload, code);
 
     return RETCODE_OK;
 }
 
 ResponseCode RTCPMessageManager::processOpenLogicalPortRequest(
-        TCPChannelResource *p_channel_resource,
+        std::shared_ptr<TCPChannelResource>& channel,
```

Comment:
Using a reference instead of a copy can cause problems as the channel being freed while we are still using it.